### PR TITLE
Curator crd for kubebuilder

### PR DESCRIPTION
**Pull request by skanthed**

### CODE REVIEW

File: cluster-scope/base/apiextensions.k8s.io/customresourcedefinitions/reports.batch.curator.openshift.io/batch.curator.openshift.io_reports.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.4.1
+  creationTimestamp: null
```
Comment (naved001) [inline cluster-scope/base/apiextensions.k8s.io/customresourcedefinitions/reports.batch.curator.openshift.io/batch.curator.openshift.io_reports.yaml:7]: looks like it expects this to be a string and not null.

https://github.com/CCI-MOC/moc-apps/runs/5131654823?check_suite_focus=true#step:4:21

is the `creationTimestamp` even required here?

Comment (skanthed) [reply cluster-scope/base/apiextensions.k8s.io/customresourcedefinitions/reports.batch.curator.openshift.io/batch.curator.openshift.io_reports.yaml]: Yes, It should be good without timestamp too. 
@naved001 updated.

File: cluster-scope/base/apiextensions.k8s.io/customresourcedefinitions/reports.batch.curator.openshift.io/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# This kustomization.yaml is not intended to be run by itself,
+# since it depends on service name and namespace that are out of this kustomize package.
+# It should be run by config/default
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- batch.curator.openshift.io_reports.yaml
+# +kubebuilder:scaffold:crdkustomizeresource
```
Comment (naved001) [inline cluster-scope/base/apiextensions.k8s.io/customresourcedefinitions/reports.batch.curator.openshift.io/kustomization.yaml:8]: what's this for? does it need to be in this file?

Comment (skanthed) [reply cluster-scope/base/apiextensions.k8s.io/customresourcedefinitions/reports.batch.curator.openshift.io/kustomization.yaml:8]: When I was deploying it on the Smaug cluster. They mentioned crd should go through Github. 
Link - https://github.com/operate-first/apps/pull/1166/files
So, I assumed the same will be the scenario for moc-staging env.

Comment (naved001) [reply cluster-scope/base/apiextensions.k8s.io/customresourcedefinitions/reports.batch.curator.openshift.io/kustomization.yaml:8]: Sorry, I wasn't clear. I'm asking what the line that begins with `#+kubebuilder:scafold` does?

Comment (surbhi0129) [reply cluster-scope/base/apiextensions.k8s.io/customresourcedefinitions/reports.batch.curator.openshift.io/kustomization.yaml:8]: Even I don’t know much but what I found is - ‘ when a controller is created, the main file has to include some code fragments to load it, and where to place those code fragments is marked by kubebuilder:scafold‘

This line is commented so will not impact the code. As this was generated by kubebuilder. There might be some commented lines.

**Review by naved001 — APPROVED** [review]

**Pull request merged**